src/main/java/org/mozilla/jss/ssl/SSLClient.java

```diff
@@ -102,13 +102,9 @@ private String getArgument(String key) {
      * the argument strings
      */
     private String cmp(String s1, String s2) {
-        if (s1 == s2)
-            return okay;
-        if (s1 == null)
+        if (s1 == null || !s1.equals(s2))
             return failed;
-        if (s1.equals(s2))
-            return okay;
-        return failed;
+        return okay;
```

Comment:
Hmm.. in the original code if both `s1` and `s2` are `null` it would return `okay`, but not in the new code. Is that intentional?

Reply:
Good spot. I think that's a bug though? We check `s1 == null` and fail if it is, so presumably the intended behaviour of `s1=s2=null` is to return `failed`?

Reply:
Looks like everywhere `cmp()` is used, the first parameter is guarded by `isInvalid()`, so it's never `null`, so we can simplify it further.

Reply:
True, that collapses this whole thing down to a one line ternary statement.